toolchain.mk

```diff
@@ -23,9 +23,11 @@ SRC_AARCH64_GCC 		?= https://developer.arm.com/-/media/Files/downloads/gnu-a/10.
 # $(3) is $.._GCC_VERSION	: the name of the file to download
 define dltc
 	@if [ ! -d "$(1)" ]; then \
-		mkdir -p $(1); \
 		echo "Downloading $(3) ..."; \
-		curl -s -L $(2) -o $(TOOLCHAIN_ROOT)/$(3).tar.xz; \
+		{ curl --retry-all-errors --retry 5 -s -L $(2) -o $(TOOLCHAIN_ROOT)/$(3).tar.xz || \
+		{ rm -f $(TOOLCHAIN_ROOT)/$(3).tar.xz; echo Download failed; exit 1; };} ; \
+		{ tar tf $(TOOLCHAIN_ROOT)/$(3).tar.xz || { rm $(TOOLCHAIN_ROOT)/$(3).tar.xz; echo Downloaded file is damaged; exit 1; };}; \
```

Comment:
Better to handle the error in `tar xf` below.

Reply:
I was thinking about this. Command "tar tf" can check archive integrity. If using "tar xf", should use "rm -rf" with variable parameter in case of errors to clean bad directory, which looks dangerous.

Reply:
I agree about trying to avoid `rm -rf $(SOME_VAR)`, but maybe we could `{ cd $(TOOLCHAIN_ROOT) && rm -rf $(3); }` which looks reasonably safe whatever value `TOOLCHAIN_ROOT` coud take.

Reply:
Done.